go-controller/pkg/clustermanager/node/node_allocator.go

```diff
@@ -634,3 +645,21 @@ func (na *NodeAllocator) hasJoinSubnetAllocation() bool {
 	// we allocate join subnets for L3/L2 primary user defined networks or default network
 	return na.netInfo.IsDefault() || (util.IsNetworkSegmentationSupportEnabled() && na.netInfo.IsPrimaryNetwork())
 }
+
+func (na *NodeAllocator) markAllocatedNetworksForUnmanagedHONode(node *corev1.Node, nodeSubnets []*net.IPNet) {
+	hostSubnet, err := houtil.ParseHybridOverlayHostSubnet(node)
+	if err != nil {
+		klog.Errorf("Failed to parse hybrid overlay for node %s: %v", node.Name, err)
+	} else if hostSubnet != nil {
+		for _, nodeSubnet := range nodeSubnets {
+			// check if the host subnet overlaps with any node subnet that is managed by the cluster subnet allocator
+			// if it does, mark it as allocated in the cluster subnet allocator
+			if utilnet.IsIPv4CIDR(nodeSubnet) == utilnet.IsIPv4CIDR(hostSubnet) && (nodeSubnet.Contains(hostSubnet.IP) || hostSubnet.Contains(nodeSubnet.IP)) {
```

Comment:
If would be very nice to have an util function that checks if two sets of networks overlap. It could most likely be based on the already available `IsContainedInAnyCIDR`